### PR TITLE
Improve message tracking and delivery receipts

### LINKED ISSUE #166: tracking: Make sure that delivery reciepts can trump error replies
Currently, the transition from ERROR to anything else is forbidden. We need to fix this to allow message delivery reciepts to trump errors from, e.g., a killed XEP-0198 session.

### PR DESCRIPTION
**Pull request by horazont**
* Include documentation for the mdr (XEP-0184) module and make it more usable.
* Add e2e tests for the mdr module.
* Fix #166.

### CODE REVIEW

File: aioxmpp/__init__.py [hunk 1]
```diff
@@ -134,6 +134,7 @@
 from .pep import PEPClient  # NOQA
 from .bookmarks import BookmarkClient  # NOQA
 from .version import VersionServer  # NOQA
+from .mdr import DeliveryReceiptsService # NOQA
```
Comment (sebastianriese) [inline aioxmpp/__init__.py]: PEP8 violation (there need to be two spaces before `#`).

File: docs/glossary.rst [hunk 2]
```diff
@@ -0,0 +1,33 @@
+Glossary
```
Comment (sebastianriese) [inline docs/glossary.rst]: Does this file really belong to this PR?

Comment (horazont) [reply docs/glossary.rst]: It was kinda "created" while writing the documentation. It is not 100% right here, but I figured "eh". Perfect is the enemy of good.

File: aioxmpp/__init__.py (hunk 1 above)
Comment (horazont) [reply aioxmpp/__init__.py]: good catch

**Pull request merged**